### PR TITLE
refactor(quiz): split oversized session-storage util + EO red-team spec (#1007, #1009)

### PR DESCRIPTION
**Pull request by okpilot**
Closes #1007. Closes #1009.

Pure concern-based refactor — identical runtime behavior and test assertions. Both files were over `code-style.md §1` size caps.

## #1009 — quiz-session-storage.ts (288L → over the 200-line util cap)
Split by concern:
- `quiz-session-storage.ts` → **199L** (active-session / localStorage: `ActiveSession`, `readActiveSession`, `buildActiveSession`, `toSessionData`, …). `readActiveSession`'s validation guards extracted into a private `isValidActiveSession()` helper to land under 200.
- new `quiz-session-handoff.ts` → **69L** (sessionStorage handoff: `sessionHandoffKey`, `SessionData`, `readSessionHandoff`, …).
- **22 importers repointed** (direct imports, no barrel; mixed-import lines split; `vi.mock` targets split so subjects-under-test still intercept the moved symbols). Tests split into `quiz-session-storage.test.ts` + `quiz-session-handoff.test.ts`.

## #1007 — get-study-questions-eo.spec.ts (595L → over the 500-line test-file cap)
- Shared setup (constants, insert/seed helpers, fixture context) extracted to `e2e/redteam/helpers/get-study-questions-eo-setup.ts` (test-infra, relaxed limit — siblings like `seed.ts`/`supabase.ts` are similar).
- `get-study-questions-eo.spec.ts` → **254L** (EO1–EO5).
- new `get-study-questions-eo-exam-oracle.spec.ts` → **120L** (EO6 mid-exam oracle + EO7 draft-status; their `quiz_sessions` seed + accumulate-then-throw cleanup moved with them).
- **Hermeticity preserved** — each spec runs its own `setupEoFixtures`/`cleanupEoFixtures` and is independently runnable. `attack-surface.md` EO vector→spec mapping updated.

## Verification
- `check-types` clean (covers the e2e specs); **441 affected unit tests pass**; lint clean.
- Red-team specs run in their own CI workflow (not runnable locally) → CI verifies the EO split's hermeticity.

## Note
The split adds one red-team spec **file**, so the suite's spec-count literal needs **+1** as a post-merge reconciliation — left untouched here because PR #1020 also bumps it (would conflict).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Improved resume flow for quiz, study, and exam sessions with more reliable handoff handling.

* **Bug Fixes**
  * Fixed session recovery so resumed sessions are stored and read consistently.
  * Strengthened question access rules so draft, deleted, cross-organization, and active-exam questions are handled correctly.

* **Tests**
  * Added broader coverage for session recovery and study-question access edge cases.

### CODE REVIEW

File: apps/web/app/app/quiz/session/_utils/quiz-session-storage.ts [hunk 1]
```diff
@@ -81,75 +75,55 @@ function safeRemove(userId: string): void {
...
+    data.currentIndex >= data.questionIds.length ||
+    typeof data.answers !== 'object' ||
+    data.answers === null ||
+    Array.isArray(data.answers)
+  ) {
+    return false
+  }
+  if (data.questionIds.some((id) => typeof id !== 'string' || !id)) return false
+  if (!isValidRecordOf(data.answers, isValidDraftAnswer)) return false
+  if (data.feedback && !isValidRecordOf(data.feedback, isValidFeedbackEntry)) return false
```
[the remark this answers is not in the document]
Comment (okpilot) [reply apps/web/app/app/quiz/session/_utils/quiz-session-storage.ts:99]: Skipping — pre-existing optional-field leniency moved verbatim by this refactor (`feedback` is optional; the `data.feedback && …` guard validates-when-present). Not introduced here, and `feedback:null` is a valid absent-feedback state downstream.

File: apps/web/e2e/redteam/helpers/get-study-questions-eo-setup.ts [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+    .update({ deleted_at: new Date().toISOString() })
+    .in('id', Array.from(createdQuestionIds))
+    .is('deleted_at', null)
+    .select('id')
+  if (error) throw new Error(`cleanupEoFixtures: ${error.message}`)
+  if ((data?.length ?? 0) > 0) {
+    console.log(`[get-study-questions-eo] soft-deleted ${data?.length} fixture question(s)`)
+  }
+  createdQuestionIds.clear()
+}
```
[the remark this answers is not in the document]
Comment (okpilot) [reply apps/web/e2e/redteam/helpers/get-study-questions-eo-setup.ts:302]: Skipping (codebase convention) — e2e helper modules are test infrastructure under the relaxed limit here; siblings `e2e/redteam/helpers/seed.ts` (666L) and `e2e/helpers/supabase.ts` (480L) far exceed 200 unflagged. Splitting to 200 diverges from the established pattern.

File: apps/web/e2e/redteam/helpers/get-study-questions-eo-setup.ts [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+    label: string
+  }): QuestionSeed => ({
+    organization_id: opts.org,
+    bank_id: opts.bank,
+    subject_id: subjectId,
+    topic_id: topicId,
+    created_by: opts.creator,
+    question_text: `${E2E_REDTEAM_EO_MARKER} ${opts.label} fixture`,
+    question_number: `${E2E_REDTEAM_EO_MARKER} ${opts.label} ${Date.now()}`,
+    explanation_text: 'Red-team EO fixture explanation.',
```
[the remark this answers is not in the document]
Comment (okpilot) [reply apps/web/e2e/redteam/helpers/get-study-questions-eo-setup.ts:228]: Skipping — `question_number` embeds the distinct `${label}` per fixture question, so same-millisecond `Date.now()` inserts don't collide (each label is unique). The red-team CI tier is green, so empirically no collision occurs.

File: apps/web/e2e/redteam/helpers/get-study-questions-eo-setup.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (okpilot) [reply apps/web/e2e/redteam/helpers/get-study-questions-eo-setup.ts:302]: Skipping — `cleanupEoFixtures` is self-cleanup of its own marker-tagged throwaway rows, not the shared-seed restoration §7's helper-test rule targets; the original inline `afterAll` was untested too. The soft-delete path is exercised by the EO red-team specs (green).
